src/main/java/io/vertx/ext/shell/term/impl/ExecCommand.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2015 Red Hat, Inc.
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ *
+ *
+ * Copyright (c) 2015 The original author or authors
+ * ------------------------------------------------------
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Apache License v2.0 which accompanies this distribution.
+ *
+ *     The Eclipse Public License is available at
+ *     http://www.eclipse.org/legal/epl-v10.html
+ *
+ *     The Apache License v2.0 is available at
+ *     http://www.opensource.org/licenses/apache2.0.php
+ *
+ * You may elect to redistribute this code under either of these licenses.
+ *
+ */
+
+package io.vertx.ext.shell.term.impl;
+
+import io.vertx.core.Handler;
+import org.apache.sshd.server.Command;
+import org.apache.sshd.server.Environment;
+import org.apache.sshd.server.ExitCallback;
+import org.apache.sshd.server.SessionAware;
+import org.apache.sshd.server.session.ServerSession;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ */
+public class ExecCommand implements Command, SessionAware, SSHExec {
+
+  private final String command;
+  private final Handler<SSHExec> handler;
+  private ExitCallback exitCallback;
+
+  public ExecCommand(Handler<SSHExec> handler, String command) {
+    this.handler = handler;
+    this.command = command;
+  }
+
+  @Override
+  public void start(Environment env) throws IOException {
+    if (handler != null) {
+      handler.handle(this);
+    } else {
+      exitCallback.onExit(0);
+    }
+  }
+
+  @Override
+  public String command() {
+    return command;
+  }
+
+  @Override
+  public void end(int status) {
+    exitCallback.onExit(status);
+  }
+
+  @Override public void setExitCallback(ExitCallback callback) {
+    this.exitCallback = callback;
+  }
+
+  @Override public void setInputStream(InputStream in) {}
+  @Override public void setOutputStream(OutputStream out) {}
+  @Override public void setErrorStream(OutputStream err) {}
```

Comment:
Sorry I am not very knowledgeable in this area, but my understanding is
that the user wants to see output when executing something like 'ssh
test@localhost -p 5000 net-ls'

I have been testing the changes, there is no error any more, but indeed no
output.

Do you mean that for now, users can only trigger actions via ssh, but not
provide input or see output?

2016-08-16 19:23 GMT+02:00 Julien Viet <notifications@github.com>:

> In src/main/java/io/vertx/ext/shell/term/impl/ExecCommand.java
> <https://github.com/vert-x3/vertx-shell/pull/27#discussion_r74980059>:
>
> > +  public String command() {
> > +    return command;
> > +  }
> > +
> > +  @Override
> > +  public void end(int status) {
> > +    exitCallback.onExit(status);
> > +  }
> > +
> > +  @Override public void setExitCallback(ExitCallback callback) {
> > +    this.exitCallback = callback;
> > +  }
> > +
> > +  @Override public void setInputStream(InputStream in) {}
> > +  @Override public void setOutputStream(OutputStream out) {}
> > +  @Override public void setErrorStream(OutputStream err) {}
>
> commands have no input/output, this is executing an SSH cli without a TTY
> session : in the future if the shell supports redirection, eventually this
> would be saved in a file or use a file as input : "ssh ... "ls -l > foo.txt"
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/vert-x3/vertx-shell/pull/27/files/b55e345db63cb5ba8621c6c3012ed8a95eb896b9#r74980059>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABbltrEAgOMCJdcfn6RX7nwcfianozqlks5qgfIOgaJpZM4JdmiF>
> .
>